iocore/net/quic/QUICFrameGenerator.cc

```diff
@@ -58,3 +59,26 @@ QUICFrameGenerator::on_frame_lost(QUICFrameId id)
     this->_info.erase(it);
   }
 }
+
+void
+QUICFrameGeneratorManager::add_generator(QUICFrameGenerator &generator, int weight)
+{
+  auto it = this->_vector.begin();
+  for (; it != this->_vector.end(); ++it) {
+    if (it->first >= weight) {
+      break;
+    }
+  }
+  this->_vector.emplace(it, weight, &generator);
+}
+
+const std::vector<QUICFrameGenerator *>
```

Comment:
Why don't you return a reference (std::vector &) ?

Reply:
Ah, because internal vector also have weights?

Because this function will be called many times, we should keep it very light. How about making a sorted vector when a new generator has been added.

Reply:
This is a good idea.